keps/sig-node/693-topology-manager/README.md

```diff
@@ -594,49 +584,299 @@ feature gate and set the CPU Manager policy to 'static'.
+manager is configured with `single-numa-node` but there are not enough resources
+available on a single numa node. 
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No. Changes in behavior only affects pods meeting the conditions scheduled after the upgrade.
+Running pods will be unaffected by any change. This offers some degree of safety in both upgrade->rollback and upgrade->downgrade->upgrade scenarios.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+Monitor the following metrics:
+
+"topology_manager_admission_requests_total"
+"topology_manager_admission_errors_total"
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The operator can look at `topology_manager_admission_requests_total` and `topology_manager_admission_errors_total`
+metrics to determine if topology manager is performing its admission check.
+In addition to that, kubelet configuration of the nodes can be inspected to check feature gates and the policies
+configured.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details: check the kubelet metric `topology_manager_admission_requests_total`
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+"topology_manager_admission_requests_total" can be used to determine if topology manager is
+performing its admission check.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Metrics
+  - Metric name:
+    - topology_manager_admission_requests_total
+    - topology_manager_admission_errors_total
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No.
+
+### Dependencies
+
+No.
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+Topology Manager is a component within kubelet for alignment of resources
+based on resource distribution across NUMA nodes and configured policy.
+Since this is a node-local feature, there are no calls to the API
+server or to the cloud provider and hence does not impact scalability.
+
+###### Will enabling / using this feature result in any new API calls?
+
+No, this is a node-local feature.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No, this is a node-local feature.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No, this is a node-local feature.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
```

Comment:
It feels strange to have no impact for scalability at any perspective especially this one. Will the resource alignment add latency comparing to without this feature?

Reply:
Let me elaborate on this. If we see this from the point of view of a pod that is requesting resources, the scheduler is responsible for placing that pod to a node. Only after the pod has been placed to a node, this feature comes into play and tries to align resources based on the policy configured on the node.

If we had 10,000 nodes in the cluster, would this feature add latency because of more nodes in the cluster? No, because once the pod is on a node, this feature is not impacted by the scale of the cluster as that is not relevant and is not factored into the alignment algorithm. It is the scheduler that has to deal with the scalability aspect and determine which out of  the 10,000 nodes can fulfill the resources requested by the pod. If this feature is turned off, the scheduler would still have to perform the same computation as it would if this feature was enabled. Hence, this feature is not impacted by scale or impacts the scalability of a cluster.

Having said that, on the node itself it perhaps adds additional latency due to the additional operations it has to do. For us to gauge that, I will add a metric related to the duration so we can have a tangible metric to evaluate the impact.

Reply:
Thanks for the clarification. I was wondering if the resource alignment gonna affect https://github.com/kubernetes/community/blob/master/sig-scalability/slos/pod_startup_latency.md and thanks for the input. It would be great to have more info provided here :)

Reply:
Thanks for the pointer, the pod startup latency is measured from pod creation timestamp to when all its containers are reported as started.

The resource alignment logic is executed at pod admission time which is prior to pod startup so there shouldn't be any impact on pod startup latency. The check at admission is to determine if the pod is suitable to be admitted on the node based on the configured policy. If considered suitable, the pod proceeds to startup (following which the pod startup latency is measured). 

I will add this info in this section.

Reply:
@cici37 I have updated the KEP with additional information consolidating all the points you highlighted. Let me know if it looks good to you.

Reply:
Oh, I see my comment above is answered here.